test/lit.cfg

```diff
@@ -877,6 +877,15 @@ if run_vendor == 'apple':
         resource_dir_opt = "-resource-dir %s" % new_resource_dir
         lit_config.note('Using freestanding resource dir: ' + new_resource_dir)
 
+    # Auto-linking does not work when stdlib is built with LTO, because linked
+    # libraries are discovered too late (after optimizations are applied). Let's
+    # explicitly link all libraries needed in tests.
+    if "stdlib_lto" in config.available_features:
```

Comment:
I think that this really needs a comment explaining that this won't work on Linux (without LLD).  The autolinking mechanism on Linux will fail due to needing `-L` arguments to the path even with this forced linking as the extracted commands will force the linker to look up the linking by name as well.

Reply:
Can you expand on this a bit? I thought we're already passing in `-L` to the resource dir. Why is Linux different from Darwin here? Forgive my ignorance, please :)

Reply:
It isn't the resource dir that is always the location for the library - the resource dir is meant for a small subset of libraries, those which are needed by the compiler directly (i.e. stdlib).  The other libraries can be anywhere.  The autolink only indicates that the library must be linked, not where to find it.  The normal lookup order/path applies for the resolution of linked libraries, even when auto-linked.  LLD will encode the `-l...` just as much as what we do now, so ultimately, the normal linker search rules apply.